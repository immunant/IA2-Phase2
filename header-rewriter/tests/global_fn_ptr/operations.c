#include "operations.h"
#include <stdio.h>

extern Op operations[2];

void call_operations(void) {
    for (int i = 0; i < 2; i++) {
        // TODO: Add a way to share strings between compartments
        //printf("%s\n", operations[i].desc.data);
        uint32_t x = 18923;
        uint32_t y = 24389;
        uint32_t res = operations[i].function(x, y);
        printf("%d = f(%d, %d)\n", res, x, y);
    }
}
