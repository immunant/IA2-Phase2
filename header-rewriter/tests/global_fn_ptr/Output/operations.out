43312 = f(18923, 24389)
461513047 = f(18923, 24389)
